web/src/presentational/Category.js

```diff
@@ -33,58 +36,83 @@ const colors = {
  * @returns {*}
  * @constructor
  */
-const Category = ({ name, category, javascriptResults }) => {
-  const { quizzes } = category;
-  return (
-    <Grid>
-      <Row>
-        <Cell
-          marginY={majorScale(3)}
-          paddingY={majorScale(2)}
-          elevation={0}
-          background={'white'}
-          columns={12}
-          justifyContent={'center'}
-        >
-          <Heading size={900} is={'h1'} textAlign={'center'}>
-            {name}
-          </Heading>
-        </Cell>
-      </Row>
-      <Row>
-        {quizzes.map((q, index) => {
-          const { border, background } = colors[javascriptResults[q.key]];
-          return (
-            <Cell
-              style={{ border: `2px solid ${border}` }}
-              background={background}
-              borderRadius={majorScale(2)}
-              height={250}
-              hoverElevation={3}
-              columns={4}
-              key={index}
-            >
-              <Link
-                style={{
-                  color: 'inherit',
-                  textDecoration: 'none',
-                  width: '100%',
-                  height: '100%',
-                  display: 'flex',
-                  justifyContent: 'center',
-                  alignItems: 'center',
-                }}
-                to={`/${category.category}/${q.key}`}
-              >
-                <Heading size={700}>{q.title}</Heading>
-              </Link>
-            </Cell>
-          );
-        })}
-      </Row>
-    </Grid>
-  );
-};
+class Category extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  handleFilter = event => {
+    const { history } = this.props;
+    const currentPath = history.location.pathname;
+    history.push(`${currentPath}?filter=${event.target.innerText}`);
+  };
+
+  render() {
+    const { name, category, javascriptResults, history } = this.props;
+    const { quizzes, subcategories } = category;
+    const parsed = queryString.parse(history.location.search);
+    return (
+      <Grid>
+        <Row>
+          <Cell
+            marginY={majorScale(3)}
+            paddingY={majorScale(2)}
+            elevation={0}
+            background={'white'}
+            columns={12}
+            justifyContent={'center'}
+          >
+            <Heading size={900} is={'h1'} textAlign={'center'}>
+              {name}
+            </Heading>
+          </Cell>
+        </Row>
+        <Row>
+          {subcategories.map(subcategory => (
+            <div key={subcategory}>
+              <button onClick={this.handleFilter}>{subcategory}</button>
+            </div>
+          ))}
+        </Row>
+        <Row>
+          {quizzes
+            .filter(q => {
+              return q.subcategory.indexOf(parsed.filter || 'all') >= 0;
```

Comment:
```suggestion
              return parsed.filter ? q.subcategory.indexOf(parsed.filter) >= 0 : true;
```
We can take out "all" subcategory. When there is nothing selected return everything

Reply:
Also in the data files

Reply:
mmm would q.subcategory.indexOf(parsed.filter || '') >= 0; work then?